1_fetch/src/get_site_info_data.R

```diff
@@ -0,0 +1,40 @@
+get_site_info_data <- function(eval_data, file_outpath){ 
+  
+  # function to access and download site information based on unique site numbers from 2_process/out/all_nwisdata.csv
+  
+    # get site list
+      nwisdata <- readr::read_csv(eval_data)
+      sites <- unique(nwisdata$site_no) 
+  
+    # download site info - loop through sites and append to a single dataframe
+      data_out <- data.frame()
+      
+      for (i in sites) {
```

Comment:
You actually don't need a `for` loop here. Running `dataRetrieval::readNWISsite(site_num)` with a vector of site numbers will return a data.frame with a row for each site. So, I think this whole function can be simplified to:

```r
get_site_info_data <- function(eval_data, file_outpath){ 

  # function to access and download site information based on unique site numbers from 2_process/out/all_nwisdata.csv

    # get site list
      nwisdata <- readr::read_csv(eval_data)
      sites <- unique(nwisdata$site_no)

      # download site info - loop through sites and append to a single dataframe
      data_out <- retry(download_nwis_site_info(sites), maxErrors = 10, sleep = 2)

      readr::write_csv(data_out, file_outpath)

      return(file_outpath)
```